src/lib/common/globals.cpp

```diff
@@ -426,24 +427,162 @@ int64_t parse8601(const std::string& s)
   return accumulated;
 }
 
+
+
+/*****************************************************************************
+*
+* timezoneOffset -
+*
+* Returns the time offset corresponding to a given timezone. Only the ISO8601 timezonoe
+* formats are supported (https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators):
+*
+* <time>Z
+* <time>±hh:mm
+* <time>±hhmm
+* <time>±hh
+*
+* The value -1 is used as "wrong timezone" (note that no timezone corresponds to an
+* offset of just one negative second)
+*
+*/
+static int timezoneOffset(const char* tz)
+{
+  // Trying the <time>Z format
+  if (strcmp(tz, "Z") == 0)
+  {
+    return 0;
+  }
+
+  // All other cases start by + or -
+  if ((tz[0] != '+') && (tz[0] != '-'))
+  {
+    return -1;
+  }
+
+  bool positive = (tz[0] == '+');
+  int offset = -1;
+  int h, m;
+
+  if (sscanf(tz + 1, "%2d:%2d", &h, &m) == 2)      // Trying the <time>±hh:mm format
+  {
+    offset = h * 60 * 60 + m * 60;
+  }
+  else if (sscanf(tz + 1, "%2d%2d", &h, &m) == 2)  // Trying the <time>±hhmm format
+  {
+    offset = h * 60 * 60 + m * 60;
+  }
+  else if (sscanf(tz + 1, "%2d", &h) == 1)         // Trying the <time>±hh format
+  {
+    offset = h * 60 * 60;
+  }
+
+  if (offset == -1)
+  {
+    // wrong timezone
+    return -1;
+  }
+
+  if (positive)
+  {
+    return offset;
+  }
+  else
+  {
+    return -offset;
+  }
```

Comment:
I would suggest a one-liner here:

    return positive? offset : -offset;

Or, make positive an int, with value 1 or -1 and call it 'sign'

    int sign = (tz[0] == '+')? 1 : -1;
￼
and then return like this:

    return sign * offset;